ts/nni_manager/common/experimentConfig.ts

```diff
@@ -207,45 +217,31 @@ export interface ExperimentConfig {
 /* util functions */
 
 const timeUnits = { d: 24 * 3600, h: 3600, m: 60, s: 1 };
+const sizeUnits = { tb: 1024 ** 4, gb: 1024 ** 3, mb: 1024 ** 2, kb: 1024, b: 1 };
 
-export function toSeconds(time: string): number {
-    for (const [unit, factor] of Object.entries(timeUnits)) {
-        if (time.toLowerCase().endsWith(unit)) {
-            const digits = time.slice(0, -1);
-            return Number(digits) * factor;
+function toUnit(value: string | number, targetUnit: string, allUnits: any): number {
+    if (typeof value === 'number') {
+        return value;
+    }
+    value = value.toLowerCase();
+    for (const [unit, factor] of Object.entries(allUnits)) {
+        if (value.endsWith(unit)) {
+            const digits = value.slice(0, -unit.length);
+            const num = Number(digits) * (factor as number);
+            return Math.ceil(num / allUnits[targetUnit]);
         }
     }
-    throw new Error(`Bad time string "${time}"`);
+    throw new Error(`Bad unit in "${value}"`);
 }
 
-const sizeUnits = { tb: 1024 * 1024, gb: 1024, mb: 1, kb: 1 / 1024 };
-
-export function toMegaBytes(size: string): number {
-    for (const [unit, factor] of Object.entries(sizeUnits)) {
-        if (size.toLowerCase().endsWith(unit)) {
-            const digits = size.slice(0, -2);
-            return Math.floor(Number(digits) * factor);
-        }
-    }
-    throw new Error(`Bad size string "${size}"`);
+export function toSeconds(time: string | number): number {
+    return toUnit(time, 's', timeUnits);
 }
 
-export function toCudaVisibleDevices(gpuIndices?: number[]): string {
-    return gpuIndices === undefined ? '' : gpuIndices.join(',');
+export function toMegaBytes(size: string | number): number {
+    return toUnit(size, 'mb', sizeUnits);
 }
 
-export function flattenConfig<T>(config: ExperimentConfig, platform: string): T {
```

Comment:
Where did you move this flattenConfig to?

Reply:
It's done in Python side.